packages/components/modal/__tests__/modal.test.tsx

```diff
@@ -91,6 +92,33 @@ describe("Modal", () => {
     expect(onClose).toHaveBeenCalled();
   });
 
+  test("should fire 'onOpenChange' callback when open button is clicked", async () => {
```

Comment:
This test case passes in canary branch which means it doesn't cover the change you added in this PR.